drivers/dac/ad3552r/ad3552r.c

```diff
@@ -474,7 +474,7 @@ int32_t ad3552r_write_reg(struct ad3552r_desc *desc, uint8_t addr,
 		buf[0] = val & 0xFF;
 	else
 		/* reg_len can be 2 or 3, but 3rd bytes needs to be set to 0 */
-		put_unaligned_be16(buf, val);
+		put_unaligned_be16(val, buf);
```

Comment:
Maybe move changes in this file to this commit:
https://github.com/analogdevicesinc/no-OS/pull/1188/commits/c1618a28e51a32d603c19c882347a91df33a13a8